docs/lifecycle-management.md

```diff
@@ -0,0 +1,185 @@
+# Higher level VM Management
+
+## Motivation
+
+So far KubeVirt provides a pod-like interface when managing VMs. This seems to
+be a good building block.
+On the other hand this might not be sufficient for usual virtualization
+management UIs, which do expect to manage the complete life-cycle of a VM,
+including shut down VMs 8which is currently not intended by the core runtime).
+
+
+This document outlines how minimal assumptions can be met and also visits how
+other requirements impact the design.
+
+
+## Requirements
+* Running and non-running VMs are handled through the same API
+* Creating a VM must be separate from launching a VM
+* Support for “flavor” based VMs (templating)
+* Support for expanding sparse VM definitions
+* Support for detailed VM definitions
+* Support hot-plugging
+* Representation of dynamic instance specific data
+* Support for defaulting (to support OS compatibility)
+Considerations
+* Sub-resources for actions
+* Presets to add certain configurations
+
+
+## Definitions
+* **Defaulting** - Setting certain defaults during expansion (i.e. virtio for a
+  bus type, or e1000 for as a nic model)
+* **Expansion** - VM definitions can be provided in a sparse way i.e. just a
+  disk. Once the VM is getting started, the runtime will infer additional
+  requirements for this device and add them to the runtime VM definition.
+  This process of inferring and adding additional devices is called expansion.
+* **Flavor** - Comparable to a template or a specific instance type
+* **Preset** - Presetting certain values in a specified path of the document
+  (i.e. setting .spec.domain.devices.nvram). It is not about setting certain
+  default values for undefined fields (i.e. set virtio for all busses)
+
+
+## Option ICID - “Imperative-Config-Instance-Dance”
+### Kinds
+
+**Template/Config**
+```yaml
+Kind: Config
+Metadata:
+  # autoSave: Auto backpropagate expansions from instance
+  autoSave: False
+Spec:
+  Template:
+    Metadata:
+    Spec:
+      Domain:
+        Devices:
+          …
+```
+
+**Instance**
+```yaml
+Kind: VM
+Metadata:
+  ownerReferences: [{$config}]
```

Comment:
To formalize this a little bit more: The field is used to indicate who owns an object in Kubernetes. The garbage collector can use this to make sure things are deleted the right way. It would reference a controller instance which created it (for instance an VMPool controller). Maybe that is what you mean?

https://github.com/kubernetes/community/blob/master/contributors/design-proposals/garbage-collection.md